src/common/snippets/src/pass/mha_tokenization.cpp

```diff
@@ -294,6 +294,11 @@ ov::snippets::pass::TokenizeMHASnippets::TokenizeMHASnippets(const SnippetsToken
 
         if (axis != rank.get_length() - 1 || interm_op->get_output_target_inputs(0).size() != 1)
             return false;
+
+        // Softmax need one buffer at least
+        if (interm_op->get_output_partial_shape(0).rbegin()->is_dynamic())
```

Comment:
What about static cases? Doesn't static Softmax require buffer as well?

Reply:
Softmax also needs one Buffer but static Buffer in Softmax always reuse reg group of Buffer after MatMul0 -> no need to check static case here.

`MM0 -> Buffer [reg_group=0] -> .... Buffer in Softmax decomp [reg_group=0] ->....`